deployer/src/deployment/state_change_layer.rs

```diff
@@ -187,60 +61,33 @@ where
         }
 
         let mut visitor = NewStateVisitor::default();
-
         attrs.record(&mut visitor);
 
-        let details = visitor.details;
-
-        if details.id.is_nil() {
+        if visitor.id.is_nil() {
             warn!("scope details does not have a valid id");
             return;
         }
 
-        // Safe to unwrap since this is the `on_new_span` method
-        let span = ctx.span(id).unwrap();
-        let mut extensions = span.extensions_mut();
-        let metadata = span.metadata();
-
-        self.recorder.record(Log {
-            id: details.id,
-            state: details.state,
-            level: metadata.level().into(),
+        // To deployer persistence
+        self.state_recorder.record_state(DeploymentState {
+            id: visitor.id,
+            state: visitor.state,
+        });
+        // To logger
+        self.log_recorder.record(LogItem {
+            id: visitor.id,
+            internal_origin: Backend::Deployer,
             timestamp: Utc::now(),
-            file: metadata.file().map(str::to_string),
-            line: metadata.line(),
-            target: metadata.target().to_string(),
-            fields: Default::default(),
-            r#type: LogType::State,
+            line: format!(" INFO Entering {:?} state", visitor.state),
         });
```

Comment:
Won't the other layer already do this?

Reply:
Oh, do you mean that each place where this state is updated, there should be a `info!` that states this line?

Reply:
I mean, won't the other layer record the `LogItem`? Meaning this layer only needs some state recorder

Reply:
The other layer does not have any sense of the state. This layer has to be the one that logs such lines.